optuna/integration/mlflow.py

```diff
@@ -0,0 +1,141 @@
+
+        .. testcleanup::
+
+            import shutil
+            shutil.rmtree(tempdir)
+
+        .. testoutput::
+            :hide:
+            :options: +NORMALIZE_WHITESPACE
+
+            INFO: 'my_study' does not exist. Creating a new experiment
+
+    Args:
+        tracking_uri:
+            Set the tracking server URI.
+
+            - An empty string, or a local directory path, prefixed with ``file:``. Data is stored
+              locally in the provided directory (or ``./mlruns`` if empty).
+            - An HTTP URI like ``https://my-tracking-server:5000``.
+            - A Databricks workspace, provided as the string ``databricks`` or, to use a
+              Databricks CLI.
+              `profile <https://github.com/databricks/databricks-cli#installation>`_,
+              ``databricks://<profileName>``.
+        experiment:
+            Name of MLflow experiment to be activated. If not set ``study.study_name``
+            will be taken. Either ``experiment`` or ``study.study_name`` must be set.
+        metric_name:
+            Name of the metric. If not provided this will be called ``trial_value``.
+    """
+
+    def __init__(self, tracking_uri=None, experiment=None, metric_name=None):
+        # type: (str, str, str) -> None
+
+        _check_mlflow_availability()
+
+        self._tracking_uri = tracking_uri
+        self._experiment = experiment
+        self._metric_name = metric_name
+
+    def __call__(self, study, trial):
+        # type: (optuna.study.Study, optuna.structs.FrozenTrial) -> None
+
+        # This sets the tracking_uri for MLflow.
+        if self._tracking_uri is not None:
+            mlflow.set_tracking_uri(self._tracking_uri)
+
+        # This sets the experiment of MLflow.
+        if self._experiment is not None:
+            mlflow.set_experiment(self._experiment)
+        elif (
+            study.study_name is not None
+            and study.study_name != "no-name-00000000-0000-0000-0000-000000000000"
+        ):
+            mlflow.set_experiment(study.study_name)
+        else:
+            raise ValueError("Either 'experiment' or 'study.study_name' must be set!")
+
+        with mlflow.start_run(run_name=trial.number):
+
+            # This sets the metric for MLflow.
+            trial_value = trial.value if trial.value is not None else float("nan")
+            metric_name = self._metric_name if self._metric_name is not None else "trial_value"
+            mlflow.log_metric(metric_name, trial_value)
+
+            # This sets the params for MLflow.
+            mlflow.log_params(trial.params)
+
+            # This sets the tags for MLflow.
+            tags = {}  # type: Dict[str, str]
+            tags["trial_number"] = str(trial.number)
+            tags["trial_datetime_start"] = str(trial.datetime_start)
+            tags["trial_datetime_complete"] = str(trial.datetime_complete)
+            tags["trial_state"] = str(trial.state)
```

Comment:
In [`Study.trials_dataframe`](https://optuna.readthedocs.io/en/latest/reference/study.html#optuna.study.Study.trials_dataframe), the common prefix, `TrialState.`, is removed by [those lines](https://github.com/optuna/optuna/blob/034cfdf1ff9cddcd9c0e5301f9a476181c5f2b41/optuna/study.py#L492-L494). To make the output more human-readable, how about taking the same approach here?

Reply:
Changed that.